wrappers/ios/Sources/Wrapper/Reader/ZXIBarcodeReader.mm

```diff
@@ -0,0 +1,109 @@
+// Copyright 2022 KURZ Digital Solutions GmbH
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#import "ZXIBarcodeReader.h"
+#import "ZXIErrors.h"
+#import "ZXing/ReadBarcode.h"
+#import "ZXing/ImageView.h"
+#import "ZXing/Result.h"
+#import "ZXIFormatHelper.h"
+
+using namespace ZXing;
+
+@interface ZXIBarcodeReader()
+@property (nonatomic, strong) CIContext* ciContext;
+@end
+
+@implementation ZXIBarcodeReader
+
+- (instancetype)init {
+    return [self initWithHints: [[ZXIDecodeHints alloc]initWithTryHarder:NO tryRotate:NO formats:@[]]];
+}
+
+- (instancetype)initWithHints:(ZXIDecodeHints*)hints{
+    self = [super init];
+    self.ciContext = [CIContext new];
+    self.hints = hints;
+    return self;
+}
+
+- (nullable ZXIResult *)readCIImage:(nonnull CIImage *)image error:(NSError **)error {
+    CGImageRef cgImage = [self.ciContext createCGImage:image fromRect:image.extent];
+    ZXIResult *result = [self readCGImage:cgImage error:error];
+    CGImageRelease(cgImage);
+    return result;
+}
+
+- (nullable ZXIResult *)readCGImage: (nonnull CGImageRef)image error:(NSError **)error {
+    CGColorSpaceRef colorSpace = CGColorSpaceCreateWithName(kCGColorSpaceGenericGray);
+    CGFloat cols = CGImageGetWidth(image);
+    CGFloat rows = CGImageGetHeight(image);
+    NSMutableData *data = [NSMutableData dataWithLength: cols * rows];
+
+    CGContextRef contextRef = CGBitmapContextCreate(
+                                                    data.mutableBytes,// Pointer to backing data
+                                                    cols,                      // Width of bitmap
+                                                    rows,                     // Height of bitmap
+                                                    8,                          // Bits per component
+                                                    cols,              // Bytes per row
+                                                    colorSpace,                 // Colorspace
+                                                    kCGBitmapByteOrderDefault); // Bitmap info flags
+
+    CGContextDrawImage(contextRef, CGRectMake(0, 0, cols, rows), image);
+    CGContextRelease(contextRef);
+
+    ImageView imageView = ImageView(
+              static_cast<const uint8_t *>(data.bytes),
+              static_cast<int>(cols),
+              static_cast<int>(rows),
+              ImageFormat::Lum);
+
+    Result result = ReadBarcode(imageView, [ZXIBarcodeReader DecodeHintsFromZXIOptions:self.hints]);
```

Comment:
Please use `ReadBarcodes()` here instead and set `DecodeHints().setMaxNumberOfSymbols(1)`. This does almost the same thing as `ReadBarcode()` but not quite for historical reasons. Also have you thought about making the wrapper return a list of multiple codes right from the start instead of adding it later on and then having to support two variants?